src/elexmodel/models/BootstrapElectionModel.py

```diff
@@ -807,8 +808,12 @@ def compute_bootstrap_errors(
         )
 
         # we use k-fold cross validation to find the optimal lambda for our OLS regression
-        optimal_lambda_y = self.cv_lambda(x_train, y_train, np.logspace(-3, 2, 20), weights=weights_train)
-        optimal_lambda_z = self.cv_lambda(x_train, z_train, np.logspace(-3, 2, 20), weights=weights_train)
+        if self.lambda_ is None:
```

Comment:
Sorry I just thought of something.  Could you add some logging here that outputs the values for `optimal_lambda_y` and `optimal_lambda_z`?  That way when we run this during testing we can grab the best `lambda_` from the logs and specify that in the config 😄

Reply:
good idea, that's done!